src/edu/ucsb/nceas/metacat/properties/PropertiesWrapper.java

```diff
@@ -140,16 +140,21 @@ protected String getProperty(String propertyName) throws PropertyNotFoundExcepti
 
         String returnVal = null;
         if (envSecretKeyMappings.containsKey(propertyName)) {
-            returnVal = System.getenv(envSecretKeyMappings.getProperty(propertyName));
+            String envKey = envSecretKeyMappings.getProperty(propertyName);
+            returnVal = System.getenv(envKey);
+            if (returnVal == null || returnVal.trim().isEmpty()) {
+                logMetacat.warn("Didn't find env secret: " + envKey
+                                    + "; falling back to property: " + propertyName);
+            } else {
+                logMetacat.info(
+                    "Using env var secret: " + envKey + " to override property: " + propertyName);
+            }
         }
-        // System.getenv(key) returns null if key not present, or empty string if defined but empty
-        // in either case, we want to try fallback:
         if (returnVal == null || returnVal.trim().isEmpty()) {
             returnVal = mainProperties.getProperty(propertyName);
         }
-        // getProperty() returning an empty value is valid, so only disallow null:
-        if (returnVal == null) {
-            logMetacat.info("did not find the property with key " + propertyName);
+        if (returnVal == null) {  // getProperty() returning an empty string is valid
+            logMetacat.error("did not find the property with key " + propertyName);
```

Comment:
Error seems right to me here. If there is a call for a property name that doesn't exist, that likely indicates either a code or config error.

Reply:
Thanks! I also struggled with the level for: `logMetacat.warn("Didn't find env secret...`. I'd be fine with making it "info" if Jing agrees (since either way, it would still require a log-level change for it to be displayed). What do you think, @taojing2002? You get the deciding vote :-)

Reply:
The `info` level sounds good to me. @artntek